### PR TITLE
Distributed logging

### PR DESCRIPTION
**Pull request by cicdw**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR fixes some issues we were seeing with logging in long-standing Dask clusters (namely, some logs didn't arrive).  In particular, this PR introduces the following behavior: whatever `prefect.context` is at _initialization_ time for the runners is guaranteed to be present during a run.  This ensures, in particular, that `flow_run_id` is always available for the logger.

### CODE REVIEW

File: src/prefect/utilities/logging.py [hunk 1]
```diff
@@ -22,14 +22,18 @@ class CloudHandler(logging.StreamHandler):
     def __init__(self) -> None:
         super().__init__()
         self.client = None
-        self.errored_out = False
+        self.logger = logger = logging.getLogger("CloudHandler")
```
Comment (jlowin) [inline src/prefect/utilities/logging.py]: Why the extra `= logger =` here? I don't think the variable is used anywhere

File: src/prefect/utilities/logging.py [hunk 2]
```diff
@@ -52,8 +56,8 @@ def emit(self, record) -> None:  # type: ignore
                 level=level,
                 info=record_dict,
             )
-        except:
-            self.errored_out = True
+        except Exception as exc:
+            self.logger.critical("Failed to write log with error: {}".format(str(exc)))
```
Comment (jlowin) [inline src/prefect/utilities/logging.py]: 👍

File: src/prefect/utilities/logging.py (hunk 1 above)
Comment (cicdw) [reply src/prefect/utilities/logging.py]: haha total oversight!

Comment (cicdw) [reply src/prefect/utilities/logging.py]: Updated.

**Pull request merged**